x/concentrated-liquidity/fees_test.go

```diff
@@ -951,3 +952,127 @@ func (s *KeeperTestSuite) TestUpdateFeeAccumulatorPosition() {
 	}
 
 }
+
+func (suite *KeeperTestSuite) TestComputeFeeChargePerSwapStep() {
+	var (
+		five              = sdk.NewDec(5)
+		belowDefaultPrice = DefaultCurrPrice.Sub(five)
+		aboveDefaultPrice = DefaultCurrPrice.Add(five)
+		onePercentFee     = sdk.NewDecWithPrec(1, 2)
+	)
+
+	sqrtBelowDefaultPrice, err := belowDefaultPrice.ApproxSqrt()
+	suite.Require().NoError(err)
+
+	sqrtAboveDefaultPrice, err := aboveDefaultPrice.ApproxSqrt()
+	suite.Require().NoError(err)
+
+	tests := map[string]struct {
+		currentSqrtPrice         sdk.Dec
+		nextTickSqrtPrice        sdk.Dec
+		sqrtPriceLimit           sdk.Dec
+		amountIn                 sdk.Dec
+		amountSpecifiedRemaining sdk.Dec
+		swapFee                  sdk.Dec
+
+		expectedFeeCharge sdk.Dec
+		expectPanic       bool
+	}{
+		"current sqrt price == next tick sqrt price -> charge fee on amount in": {
+			currentSqrtPrice:         DefaultCurrSqrtPrice,
+			nextTickSqrtPrice:        DefaultCurrSqrtPrice,
+			sqrtPriceLimit:           sqrtAboveDefaultPrice,
+			amountIn:                 sdk.NewDec(100),
+			amountSpecifiedRemaining: five,
+			swapFee:                  onePercentFee,
+
+			expectedFeeCharge: sdk.OneDec(),
+		},
+		"current sqrt price == sqrt price limit -> charge fee on amount in": {
+			currentSqrtPrice:         DefaultCurrSqrtPrice,
+			nextTickSqrtPrice:        sqrtAboveDefaultPrice,
+			sqrtPriceLimit:           DefaultCurrSqrtPrice,
+			amountIn:                 sdk.NewDec(100),
+			amountSpecifiedRemaining: five,
+			swapFee:                  onePercentFee,
+
+			expectedFeeCharge: sdk.OneDec(),
+		},
```

Comment:
Im a little confused here, if the currentPrice is equal to the limit and the next tick is upwards, shouldn't the swap never go through?

Reply:
If we hit the price limit, we are still allowed to consume some `amountIn` during the swap. It would have been a larger amount if not the price limit. However, we still need to charge the fee on some `amountIn` consumed up until we hit the price limit